### ISSUE TITLE
`value-of` in array-shapes do not properly resolve

### ISSUE THREAD

**Issue by boesing** [vimeo/psalm#9282]
Hey there,

I try to do an array shape assertion. Somehow, psalm does not properly handle `value-of<Enum>` within that array shape and thus, I end up getting an issue.
It seems to work when directly asserting the value with `value-of<Enum>` tho and thus, I guess there might be some codepath missing to resolve `value-of` properly.

https://psalm.dev/r/eba33fb727

**Comment by weirdan** [comment]
@boesing could this have been resolved (or partially resolved) by your recent PR?

**Comment by boesing** [comment]
Yeah, at least the `InvalidArgument` to `RelationEnum::from` is fixed.
